spec/jasmine/frameSpec.js

```diff
@@ -0,0 +1,131 @@
+describe("FrameNotTen", function () {
+  describe("Points tests", function () {
+    let frame;
+    beforeEach(function () {
+      frame = new FrameNotTen(3, 0);
+    });
+
+    it("a new frame should have the frame score 0", function () {
+      expect(frame.points("score")).toEqual(0);
+    });
+
+    it("a new frame should have the frame bonus 0", function () {
+      expect(frame.points("bonus")).toEqual(0);
+    });
+
+    it("a new frame should have the total score 0", function () {
+      expect(frame.points()).toEqual(0);
+    });
+
+    it("a new frame should have the total score 3", function () {
+      frame._addRollAndReturnBonusForPrevFrames(3);
+      expect(frame.points()).toEqual(3);
+    });
+
+    it("frame after adding bonus should have the bonus 3", function () {
+      frame.addBonusSentFromFutureRolls(3);
+      expect(frame.points("bonus")).toEqual(3);
+    });
+
+    it("frame after adding bonus and score should have the total 7", function () {
+      frame._addRollAndReturnBonusForPrevFrames(3);
+      frame.addBonusSentFromFutureRolls(4);
+      expect(frame.points()).toEqual(7);
+    });
+  });
+
+  describe("test _addRollAndReturnBonusForPrevFrames return bonusPrevFrames", function () {
```

Comment:
write test descriptions in plain English as much as possible so the Jasmine output reads nicely